packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4708,11 +4721,21 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   /// Implementations of this method should end with a call to the inherited
   /// method, as in `super.deactivate()`.
   @mustCallSuper
+  @visibleForOverriding
```

Comment:
I added the annotation because I was trying to move `_ensureDeactivated` out, but since nothing is failing I'll keep the annotation (same as activate).